src/storage/heap_file.c

```diff
@@ -22472,16 +22424,6 @@ heap_update_relocation (THREAD_ENTRY * thread_p, HEAP_OPERATION_CONTEXT * contex
    */
   if (update_old_forward)
     {
-      /* 
-       * Updating forward record and setting prev_version_lsa have to be atomic
-       * if the home record (REC_RELOCATION) is not updated.
-       */
-      if (!atomic_replication_flag && is_mvcc_op)
-	{
-	  log_append_empty_record (thread_p, LOG_START_ATOMIC_REPL, NULL);
-	  atomic_replication_flag = true;
-	}
-
```

Comment:
We have to update a record and its previous version lsa (RVHF_SET_PREV_VERSION_LSA) atomically. It requires atomic operation. There are some cases. Please see 9cd23bcc8f.

Reply:
Although some cases still require atomic operation because of RVHF_SET_PREV_VERSION_LSA, I think we can shrink atomic operation's range even for them.

Reply:
OK, I didn't take RVHF_SET_PREV_VERSION_LSA into account. Thanks. I will consider that case and update jira issue.